Riot/Modules/Room/ParticipantsInviteModal/ContactsPicker/ContactsPickerViewModel.swift

```diff
@@ -0,0 +1,282 @@
+        // If the homeserver has converted the 3pid invite into a room member, do no show it
+        // If the invite has been revoked (null display name), do not show it too.
+        guard let displayName = invite.displayname, roomState.member(withThirdPartyInviteToken: invite.token) == nil else {
+            return
+        }
+        
+        if let contact = Contact(matrixContactWithDisplayName: displayName, andMatrixID: nil) {
+            contact.isThirdPartyInvite = true
+            contact.mxThirdPartyInvite = invite
+            members.invitedParticipants.append(contact)
+        }
+    }
+    
+    private func finalize(participants roomMembers: RoomMembers) {
+        self.actualParticipants = roomMembers.actualParticipants
+        self.invitedParticipants = roomMembers.invitedParticipants
+        self.userParticipant = roomMembers.userParticipant
+        self.coordinatorDelegate?.contactsPickerViewModelDidEndLoading(self)
+    }
+}
+
+// MARK: - ContactsTableViewControllerDelegate
+extension ContactsPickerViewModel: ContactsTableViewControllerDelegate {
+    
+    func contactsTableViewController(_ contactsTableViewController: ContactsTableViewController!, didSelect contact: MXKContact?) {
+        guard let contact = contact else {
+            MXLog.error("[ContactsPickerViewModel] contactsTableViewController: nil contact found")
+            return
+        }
+        
+        let roomName = room.displayName ?? VectorL10n.spaceTag
+        let message = VectorL10n.roomParticipantsInvitePromptToMsg(contact.displayName, roomName)
+        
+        coordinatorDelegate?.contactsPickerViewModel(self, display: message, title: VectorL10n.roomParticipantsInvitePromptTitle, actions: [
+            UIAlertAction(title: MatrixKitL10n.cancel, style: .cancel, handler: nil),
+            UIAlertAction(title: VectorL10n.invite, style: .default, handler: { [weak self] action in
+                self?.invite(contact: contact)
+            })
+        ])
+    }
+    
+    private func invite(contact: MXKContact) {
+        if let identifiers = contact.matrixIdentifiers as? [String], let participantId = identifiers.first {
+
+            // Invite this user if a room is defined
+            self.coordinatorDelegate?.contactsPickerViewModelDidStartInvite(self)
+            room.invite(.userId(participantId)) { [weak self] response in
+                guard let self = self else { return }
+                
+                switch response {
+                case .success:
+                    self.coordinatorDelegate?.contactsPickerViewModelDidEndInvite(self)
+                case .failure:
+                    MXLog.error("[ContactsPickerViewModel] Failed to invite \(participantId) due to error; \(response.error ?? "nil")")
+                    AppDelegate.theDelegate().showError(asAlert: response.error)
+                }
+            }
+        } else {
+            let _participantId: String?
```

Comment:
I prefixed this one to avoid linter error for the guard bellow:
```
            guard let participantId = _participantId else {
                MXLog.error("[ContactsPickerViewModel] invite: unexpectedly found participantId nil")
                return
            }
```

Reply:
*sigh* we have some silly linting rules! Never mind :)